kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigSecret.java

```diff
@@ -119,22 +120,23 @@ String generateProxyConfig(KafkaProxy primary,
     }
 
     @NonNull
-    private static LinkedHashMap<String, VirtualCluster> buildVirtualClusters(KafkaProxy primary, Context<KafkaProxy> context, List<Clusters> clusters) {
+    private static LinkedHashMap<String, VirtualCluster> buildVirtualClusters(KafkaProxy primary, Context<KafkaProxy> context, List<VirtualKafkaCluster> clusters) {
+        AtomicInteger clusterNum = new AtomicInteger(0);
         var virtualClusters = clusters.stream()
                 .filter(cluster -> !SharedKafkaProxyContext.isBroken(context, cluster))
                 .collect(Collectors.toMap(
-                        Clusters::getName,
-                        cluster -> getVirtualCluster(primary, cluster),
+                        (cluster) -> cluster.getMetadata().getName(),
```

Comment:
I think cluster is auto generated so there is not much we can do, but I I'm not a fan of callers needing to know that the name is stored in metadata.

Reply:
yeap it's a bit crap, pushes us to maintain a spaghetti of static methods if we want re-use. Maybe it's worth having a light set of records that we wrap around them. Like `RichKafkaCluster.wrap(KafkaCluster primary)`.

Reply:
for another PR of course